### PR TITLE
docs(website): add theme-environment-variables community plugin

### PR DESCRIPTION
**Pull request by dnguyen0304**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have signed the [Contributor License Agreement](https://docusaurus.io/community/contributing#contributor-license-agreement-cla).

## Motivation

I'd like to add a link to `theme-environment-variables` on the community resources page.

This was originally discussed on Discord [here](https://discord.com/channels/398180168688074762/877243736806019102/1085640344449851392).

You'll note I also modified `theme-github-codeblock`. The convention on this page appears to use hyphens as delimiters but this list item used a period. I can remove this change from the PR if you'd like.

## Test Plan

Screen Shot 2023-03-20 at 7 53 45 AM

### Test links

Deploy Preview: https://deploy-preview-8793--docusaurus-2.netlify.app/community/resources/

### CODE REVIEW

**Comment by slorber** [comment]
Hey

That looks cool

Was wondering why you call it "theme-environment-variables", I find this name confusing for our community considering it only applies to code blocks while intuitively I'd expect it to work in Markdown files and React components.

I mean `process.env.VAR_NAME` looks like the most common use-case

**Comment by dnguyen0304** [comment]
**TL;DR:**
- fix workflow error
- +1 to renaming the plugin
- +1 to supporting a non-sidebar UI

**Changes:**
- I added a commit to fix the "unknown word" lint error.
- I also fixed the PR title to match the Docusaurus git message convention. Apologies for the mistake.

**Topic - Rename:**
- That makes a lot of sense! I really like your suggestion of the "codeblock" namespace. That makes it clear that this plugin doesn't have anything to do with normal shell environment variables.
- Because it will require changing code references, do you mind if we make this change in a follow-up PR?
- Here is the tracking issue: https://github.com/docupotamus/docupotamus/issues/36

**Topic - Non-Sidebar UI:**
- Yes, currently it's only possible to change the code blocks using the sidebar UI. I agree it should be possible to update without the sidebar UI, but I haven't gotten around to it yet.
- Here is the tracking issue: https://github.com/docupotamus/docupotamus/issues/35

**Topic - Use Case:**

Screen Shot 2023-03-24 at 4 56 47 AM

- You know how the Docusaurus Getting Started (https://docusaurus.io/docs/installation#scaffold-project-website) defaults to "my-website" as the site name? A use case is this plugin can parameterize the site name so users can personalize the code block. This is more pleasant than copying and updating the snippet before running it. This is also why the use case doesn't work as a static plugin option.

**Comment by slorber** [comment]
@dnguyen0304 

> Because it will require changing code references, do you mind if we make this change in a follow-up PR?

I don't understand what you mean  here but I'd prefer to merge this PR only once you have updated your project's name already. Why add it with the wrong name and then submit another PR just after?

**Review by Sanjay-shankar — COMMENTED** [review]
This looks cool

**Comment by slorber** [comment]
@dnguyen0304 I'm closing due to inactivity, feel free to ping me if you update your branch/PR with an updated name
